hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -242,6 +244,8 @@ public HazelcastClientInstanceImpl(ClientConfig config,
         lockReferenceIdGenerator = new ClientLockReferenceIdGenerator();
         nearCacheManager = clientExtension.createNearCacheManager();
         clientExceptionFactory = initClientExceptionFactory();
+
+        statistics = new Statistics(this);
```

Comment:
I know the problem. In some parts on the serverside we also have this issue where 'this' is being passed to constructor.